en/guide/message_signing.md

```diff
@@ -1,121 +1,157 @@
+> **Note** The [incompatibility flags](../guide/mavlink_2.md#incompat_flags) in the packet header are used to indicate that the MAVLink library must reject the packet if it does not understand or cannot handle the flag.
+  In other words, a MAVLink library that does not support signing must drop signed packets.
+  The C library uses [MAVLINK_IFLAG_SIGNED](../guide/mavlink_2.md#MAVLINK_IFLAG_SIGNED) to represent the "supports message signing" bit.
 
-* The current timestamp should be stored regularly in persistent storage (suggested at least once a minute)
-* The timestamp used on startup should be the maximum of the timestamp implied by the system clock and the stored timestamp
-* If the system does not have a RTC mechanism then it should update its timestamp when GPS lock is achieved. The maximum of the timestamp from the GPS and the stored timestamp should be used
-* The timestamp should be incremented by one on each message send. This is done for you by the generated headers.
-* When a correctly signed message is decoded the timestamp should be replaced by the timestamp of the incoming message if that timestamp is greater than the current timestamp. This is done for you by the generated headers
-* The timestamp on incoming signed messages should be checked against the previous timestamp for the incoming `(linkID,srcSystem,SrcComponent)` tuple and the message rejected if it is smaller. This is done for you by generated headers.
-* If there is no previous message with the given `(linkID,srcSystem,SrcComponent)` then the timestamp should be accepted if it not more than 6 million (one minute) behind the current timestamp
+The 13 bytes of the signature are:
 
-## Enabling Signing on a Channel
+Data | Description
+--- | ---
+[linkID](#link_ids) (8&nbsp;bits) | ID of link on which packet is sent. Normally this is the same as the *channel*. 
+[timestamp](#timestamps) (48 bits) | Timestamp in 10 microsecond units since 1st January 2015 GMT time. This *must* monotonically increase for every message on a particular [link](#link_ids). Note that means the timestamp may get ahead of the actual time if the packet rate averages more than 100,000 packets per second.
+[signature](#signature) (48 bits) | A 48 bit signature for the packet, based on the complete packet, timestamp, and secret key.
 
-To enable signing on a channel you need to fill in two pointers in the status structure for the channel. The two pointed are:
+See below for more information about the fields.
 
-```
-mavlink_signing_t *signing;
-mavlink_signing_streams_t *signing_streams;
-```
+### Link IDs {#link_ids}
 
-The `signing` pointer controls signing for this stream. It is per-stream, and contains the secret key, the timestamp and a set of flags, plus an optional callback function for accepting unsigned packets. Typical setup would be:
+The 8 bit link ID is provided to ensure that the signature system is robust for multi-link MAVLink systems. 
```

Comment:
Just to confirm, this link ID is used to change the signature so that the same message sent over a different link does not match - ie it prevents cross-link replay. But the link ID isn't actually read/parsed by anything except for the case discussed in mavgen C library where you want to slave link IDs to each other. Correct?